### DISCUSSION TITLE
How to land user to 'last read message' when entering a chat room

### DISCUSSION THREAD

**Discussion by jisusushi** [GetStream/stream-chat-android#2441] [Q&A]
I want to send users to 'last read message' position when they're entering a chat room. I'm currently using lazyColumn for showing message list, and I know I can assign fisrt visible index in lazyColumn's listState (and can get the right index by `unreadCount`). But, to do so, if the unreadCount is like 400, I have to load 400 messages by `ChatDomain` first, and it is really annoying. 
Is their any other way to handle this problem, especially when the unreadCount is huge? I found the function `loadMessageById`, but I've got no idea how to get 92th message's Id without loading the whole message after it.

**Comment by filbabic** [comment]
Hey @jisusushi !

Thank you for reaching out!

The `unreadCount` within `MessagesState` represents the number of messages the user hasn't read while scrolling in an active chat. In that sense, all of the unread messages will be loaded in the list, if you're connecting to our API using our ViewModels.

So all you have to do is access your `LazyListState` and call `lazyListState.animateScrollToItem(index)` or `scrollToItem()` and give it the index of the last read message.

If you're thinking of scrolling to a position when opening the chat, you can use the `channel.read` status to see what the last message/date for a given user is and then you'll have to load messages up to that point and then perform the scroll.

Let me know if this works and clears up your concerns!

**Comment by workspace** [comment]
I think these changes help this discussion.
https://github.com/GetStream/stream-chat-android/pull/3472 (focus to specific message)
https://github.com/GetStream/stream-chat-android/pull/3557 (infinite scroll)
